src/main/java/org/broadinstitute/hellbender/tools/spark/pathseq/PSFilter.java

```diff
@@ -0,0 +1,334 @@
+                .partitionBy(new HashPartitioner(reads.getNumPartitions()))
+                .map(Tuple2::_2);
+    }
+
+    /**
+     * Maps partition to a Tuple of two Lists, the first containing the paired reads, the second containing unpaired
+     */
+    static Tuple2<List<GATKRead>, List<GATKRead>> getPairedAndUnpairedLists(final Iterator<GATKRead> iter, final int readsPerPartitionGuess) {
+        //Find the paired and unpaired reads by scanning the partition for repeated names
+        final List<GATKRead> pairedReadsList = new ArrayList<>(readsPerPartitionGuess);
+        final Map<String, GATKRead> unpairedReads = new HashMap<>(readsPerPartitionGuess);
+        while (iter.hasNext()) {
+            final GATKRead read = iter.next();
+            final String readName = read.getName();
+            //If read's mate is already in unpairedReads then we have a pair, which gets added to the ordered List
+            if (unpairedReads.containsKey(readName)) {
+                pairedReadsList.add(read);
+                pairedReadsList.add(unpairedReads.remove(readName));
+            } else {
+                unpairedReads.put(readName, read);
+            }
+        }
+        //Get the unpaired reads out of the hashmap
+        final List<GATKRead> unpairedReadsList = new ArrayList<>(unpairedReads.values());
+
+        //Minimize unpairedReads memory footprint (don't rely on readsPerPartitionGuess)
+        final List<GATKRead> pairedReadsListResized = new ArrayList<>(pairedReadsList.size());
+        pairedReadsListResized.addAll(pairedReadsList);
+
+        return new Tuple2<>(pairedReadsListResized, unpairedReadsList);
+    }
+
+    /**
+     * Preferentially filters unpaired reads, when possible. Assumes reads have pairedness flags set properly
+     */
+    @VisibleForTesting
+    static JavaRDD<GATKRead> filterDuplicateSequences(final JavaRDD<GATKRead> reads) {
+        return reads.mapToPair(PSFilter::canonicalizeRead)
+                .groupByKey()
+                .values()
+                .map(iter -> {
+                    for (final GATKRead read : iter) {
+                        if (!read.isPaired()) return read;
+                    }
+                    return iter.iterator().next();
+                });
+    }
+
+    /**
+     * Pairs reads with canonical Long ID using the lesser 64-bit hash of the sequence and its reverse complement
+     */
+    @VisibleForTesting
+    static Tuple2<Long, GATKRead> canonicalizeRead(final GATKRead read) {
+        final byte[] bases = read.getBases();
+        final long hashForward = fnvByteArray64(1099511628211L, bases);
+        SequenceUtil.reverseComplement(bases);
+        final long hashReverse = fnvByteArray64(1099511628211L, bases);
+        return new Tuple2<>(Math.min(hashForward, hashReverse), read);
+    }
+
+    /**
+     * 64-bit FNV-1a hash for byte arrays
+     */
+    private static long fnvByteArray64(long start, final byte[] toHash) {
```

Comment:
Maybe stick this into SVUtils with the other FNV methods? and provide a one-arg overload that encapsulates the 1099511628211L?